app/js/ellipsis.js

```diff
@@ -73,6 +73,12 @@ angular.module('sn.ellipsis', [
             testEl.style.height = 'auto';
             testEl.style.width = elementWidth + 'px';
 
+            // apply computed font/padding/margin styles to ensure test element
+            // is properly comparable with its visible equivalent
+            testEl.style.fontSize = window.getComputedStyle($element[0]).fontSize;
```

Comment:
looks good only one minor issue, for ng1 would recommend using `$window` instead of `window` as it makes it easier to test.